components/Feed/Post/PostHeader.tsx

```diff
@@ -27,21 +27,37 @@ const PostHeader: FC<PostHeaderProps> = ({
     // router from next.js to use location functions.
     const router = useRouter()
 
-    // Deletes a post
-    const deletePost = async () => {
-        // Get the path to the post to delete
-        const path = `posts/${id}`
-
-        // Get the delete function and call it on the path
-        const functions = getFunctions()
-        const deleteFn = httpsCallable(functions, 'recursiveDelete')
-        deleteFn({ path: path }).catch(function (err) {
-            console.log('Delete failed, see console,')
-            console.warn(err)
+    // Delete the post entry from the DB.
+    // Note: this post should NOT have any comments
+    const deletePostEntry = async () => {
+        const postDocRef = doc(db, 'posts', id)
+        await deleteDoc(postDocRef).catch((err) => {
+            console.log('Cannot delete post: ', err)
         })
 
         // Delete the post's media, if any
         deleteMedia(`posts/${id}`)
+    }
+
+    // Deletes a post
+    const deletePost = async () => {
+
+        let activitiesQuery = query( 
+            collection(db, "post-activity"), 
+            where("postId", '==', id)
+        )
+
+        getDocs(activitiesQuery).then(async (sub) => {
+            sub.forEach((activity) => {
```

Comment:
We still have a forEach here, can't we use the function above given the path to the post? Or are you saying that i not possible because the id of the post is different from postId? If that is the case, I would force the doc to have the same id as the argument of this class 'id'.

Reply:
In this part, we are aiming to delete all the comments and replies corresponding to the post to be deleted.
In firebase, if we want to delete, let's say all the 10 documents returned in the resultSet of a query, there's no better way than iterating through those 10 documents and calling delete on each ref. 

In comparison to our schema's performance, the complexity of deleting a post would be proportional only to number of (posts + comments) and would be independent of the size of our collection.

Let me know if I'm missing something.

Reply:
sounds good